src/activity_handlers.cpp

```diff
@@ -517,12 +495,16 @@ static void set_up_butchery( player_activity &act, player &u, butcher_type actio
             }
         } else {
             u.add_msg_if_player( m_good, _( "It needs a coffin, not a knife." ) );
-            act.index = -1;
+            act.targets.pop_back();
             return;
         }
     }
 
     act.moves_left = butcher_time_to_cut( u, corpse_item, action );
+
+    // We have a valid target, so preform the full finish function
+    // instead of just selecting the next valid target
+    act.index = false;
```

Comment:
IS `act.index` a `bool`?  It seems unlikely...

Reply:
Ah, I see later that you are treating it as a `bool`.  I can't decide whether it's clearer to assign `0` or `false` here. Hmm.

Reply:
Yeah I agree that it's not super clean.  I considered adding either a `bool` or an `std::vector<bool>` to `player_activity` but decided that was outside the scope of this PR

Reply:
I have an aspiration to make accessing things inside `player_activity` cleaner in general, with some vague ideas of how to go about it.  Looking at your code here makes me want to get to that, but I probably won't soon.